server/src/controllers/organizations.controller.ts

```diff
@@ -30,11 +32,43 @@ export class OrganizationsController {
     private readonly configService: ConfigService
   ) {}
 
-  @UseGuards(JwtAuthGuard)
+  @UseGuards(JwtAuthGuard, IsAdminGuard)
   @Get('users')
-  async getUsers(@Request() req) {
-    const result = await this.organizationsService.fetchUsers(req.user);
-    return decamelizeKeys({ users: result });
+  async getUsers(@User() user, @Query() query) {
+    const { page, email, firstName, lastName } = query;
+    const filterOptions = {
+      ...(email && { email }),
+      ...(firstName && { firstName }),
+      ...(lastName && { lastName }),
+    };
+    const usersCount = await this.organizationsService.usersCount(user, filterOptions);
+    let users = [];
+    if (usersCount > 0) users = await this.organizationsService.fetchUsers(user, page, filterOptions);
+
+    const meta = {
+      total_pages: Math.ceil(usersCount / 10),
+      total_count: usersCount,
+      current_page: parseInt(page || 1),
+    };
+
+    const response = {
+      meta,
```

Comment:
Probably need to decamelize the meta key also? wdyt?

Reply:
Sure